src/Controls/src/Core/Compatibility/Handlers/Shell/Android/ShellToolbarTracker.cs

```diff
@@ -561,15 +561,15 @@ void UpdateNavBarHasShadow(Page page)
 			if (page == null || !_appBar.IsAlive())
 				return;
 
+			// 4 is the default
+			_appBarElevation = _appBar.Context.ToPixels(4);
```

Comment:
The point of the original code was to reduce the amount of calls to `_appBar.Context`. The original code did a really bad job of it :-) 

Maybe something like this? I think this is what the original author was intending.

```C#
			if (Shell.GetNavBarHasShadow(page))
			{
				if (_appBarElevation <= 0)
				     _appBarElevation = _appBar.Context.ToPixels(4);

				_appBar.SetElevation(_appBarElevation);
			}
			else
			{
				// 4 is the default
				_appBar.SetElevation(0f);
			}
```

Reply:
Changed, thanks!